plugins/startpage.js

```diff
@@ -0,0 +1,182 @@
+        // If value not found then return []
+        // Search is NOT case-sensitive
+        // ref: https://gist.github.com/killants/569c4af5f2983e340512916e15a48ac0
+        function getValuesInJsonObject(jsonObj, searchValue, isRegex, isPartialMatch, isFirstMatchOnly, maxDeepLevel, currDeepLevel) {
+
+            var bShowInfo = false;
+
+            maxDeepLevel = ( maxDeepLevel || maxDeepLevel == 0 ) ? maxDeepLevel : 100;
+            currDeepLevel = currDeepLevel ? currDeepLevel : 1 ;
+            isRegex = isRegex ? isRegex : false;
+
+            // check RegEx validity if needed
+            var re;
+            if ( isRegex ) {
+                try {
+                    re = new RegExp(searchValue);
+                } catch (e) {
+                    cLog(e);
+                    return [];
+                }
+            } else {
+                searchValue = searchValue.toString().toLowerCase();
+            }
+
+            if ( currDeepLevel > maxDeepLevel ) {
+                return [];
+            } else {
+
+                var keys = [];
+
+                for ( var curr in jsonObj ) {
+                    var currElem = jsonObj[curr];
+
+                    if ( currDeepLevel == 1 && bShowInfo ) { cLog("getValuesInJsonObject : Looking property \"" + curr + "\" ") }
+
+                    if ( typeof currElem == "undefined" ) continue;
+
+                    if ( typeof currElem == "object" ) { // object is "object" and "array" is also in the eyes of "typeof"
+                        // search again :D
+                        var deepKeys = getValuesInJsonObject( currElem, searchValue, isRegex, isPartialMatch, isFirstMatchOnly, maxDeepLevel, currDeepLevel + 1 );
+                        for ( var e = 0; e < deepKeys.length; e++ ) {
+                            // update path backwards
+                            deepKeys[e].path = '["' + curr + '"]' + deepKeys[e].path;
+                            keys.push( deepKeys[e] );
+                        }
+                    } else {
+
+                        if ( isRegex ? re.test(currElem) : ( isPartialMatch ? currElem.toString().toLowerCase().indexOf(searchValue) != -1 : currElem.toString().toLowerCase() === searchValue ) ){
+
+                            var r = {};
+                            r.key = curr;
+                            r.value = currElem;
+                            r.path = '["' + curr + '"]';
+                            r.depth = currDeepLevel;
+                            keys.push( r );
+                            if (isFirstMatchOnly) return keys;
+                        }
+                    }
+                }
+                return keys;
+            }
+        }
+
+        // Return JSON object corresponding to path, without using the Evil eval
+        // path syntax: [key1][key2][key3]...
+        function getJsonObjectFromPath(path) {
+            return new Function('return ' + spJson + path)();
```

Comment:
@GrosPoulet This is unsafe code which is banned by Mozilla, please take a look at my fix in https://github.com/extesy/hoverzoom/commit/98a6d975a24d2aef31209e8d3fdd962c045d6c18

Reply:
@extesy I don't see what's wrong but i trust Mozilla, they are certainly more experts that i am :smile:
I'll apply your changes to StartPage plug-in & move following functions to hoverzoom.js:
- getJsonObjectFromPath
- getKeysInJsonObject
- getValuesInJsonObject

Reply:
Yeah, it makes sense to move these functions to a common code since they are used across multiple plugins.